oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -412,6 +405,7 @@ impl Runtime {
         // add itself again. Note that since that lock is currently held, the woken thread will add
         // itself to waiting_threads *after* we call clear below as we release the lock implicilty
         // on leaving this function.
+        let mut waiting_threads = channel.waiting_threads.lock().unwrap();
```

Comment:
(optional) Perhaps an explicit anonymous scope with a comment here may be useful to clarify what the scope of the lock is, especially if in the future some more instructions are added after `waiting_threads.clear()`.

Reply:
I'm [planning](https://github.com/project-oak/oak/issues/779#issuecomment-607170580) on killing the `waiting_threads` mutex soon anyhow &ndash; just need to untangle one place that takes the same lock twice (which is (apparently) OK for `RwLock::read` but goes wrong if one has to be converted to a `RwLock::write`).